sim/paladin/consecration.go

```diff
@@ -4,20 +4,30 @@ import (
 	"time"
 
 	"github.com/wowsims/tbc/sim/core"
+	"github.com/wowsims/tbc/sim/core/proto"
 	"github.com/wowsims/tbc/sim/core/stats"
 )
 
-const SpellIDConsecration int32 = 27173
+func (paladin *Paladin) registerConsecrationSpell(sim *core.Simulation) {
+	cdTimer := paladin.NewTimer()
+	paladin.registerConsecrationSpellRank6(sim, cdTimer)
+	paladin.registerConsecrationSpellRank4(sim, cdTimer)
+	paladin.registerConsecrationSpellRank1(sim, cdTimer)
 
-var ConsecrationActionID = core.ActionID{SpellID: SpellIDConsecration}
+}
 
-func (paladin *Paladin) registerConsecrationSpell(sim *core.Simulation) {
+const ConsecrationCDTime = time.Second * 8
+const SpellIDConsecrationRank6 int32 = 27173
+
+var ConsecrationRank6ActionID = core.ActionID{SpellID: SpellIDConsecrationRank6}
+
+func (paladin *Paladin) registerConsecrationSpellRank6(sim *core.Simulation, cdTimer *core.Timer) {
```

Comment:
Seems like you could refactor registerConsecration1/4/6 into a single function by adding a couple extra parameters (actionID, baseDamage, etc)